autoload/youcompleteme.vim

```diff
@@ -738,6 +738,20 @@ endfunction
 
 command! YcmShowDetailedDiagnostic call s:ShowDetailedDiagnostic()
 
+function! s:FixIt()
+  " We force recompilation to ensure that the FixIt diagnostic data is not out
+  " of date with respect to the code. Without this we can get unpredicable
+  " results.
+  "
+  " note we do this silently so that we don't echo 2 messages vim. This ensures
+  " that this command doesn't require the user to hit return to dismiss the
+  " Fixit made <N> changes message
+  if s:ForceCompileSilent()
+    py ycm_state.FixIt()
+  endif
+endfunction
+
+command! YcmFixIt call s:FixIt()
```

Comment:
The challenge is that FixIt in clang are associated with diagnostics. ycmd returns diagnostics as the response to OnFileReadyToParse and it seemed trivial to return the fixit info (in an editor agnostic way) as part of that response. So we return the fixit data with the diagnostic and the editor applies it when requested. I suppose we could add a client/server hop instead to return fixits for a given line, and create a command/response structure. To me that seems more moving parts than are necessary and might introduce unwelcome latency. At present (afaik) there is no mechanism to get diagnostic data from the server  other than the OnFileReadyToParse callback, and i didn't want to invent too much new stuff :)